text/0002-modules.md

```diff
@@ -0,0 +1,59 @@
+# Modules
+- Start Date: 2019/04/29
+- RFC PR: https://github.com/openmrs/openmrs-rfc-frontend/pull/2
+
+## Decision, including impact on distributions
+In-browser Javascript modules are the OpenMRS mechanism for adding functionality and creating frontend code for distributions.
+Since modules are not yet fully supported in all browsers, OpenMRS uses SystemJS as its in-browser module loader.
+
+## Definition
+A [javascript module](https://www.sitepoint.com/understanding-es6-modules/) is a new kind of javascript file that
+can [import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) and
+[export](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export) data and logic from other modules.
+Variables within a module are not global-by-default and are not accessible to other javascript code unless explicitly
+exported.
+
+Javascript modules are also known as ECMAScript modules, ES modules, ESM, ES2015 modules, or ES6 modules.
+
+An in-browser javascript module is a javascript module that is downloaded in the browser. This is different from
+build-time modules, which are compiled down so that they know longer use `import` or `export` by the time they
+are executed in the browser. Most major frameworks (React, Vue, Angular, etc) always use build-time modules, but do not
+always use in-browser modules. At OpenMRS, we use both.
+
+In-browser javascript modules are downloaded with one network request per module. Build-time modules are compiled away such
+that there is only one network request for a whole group (bundle) of build-time modules.
+
+[Basic browser support for modules](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export#Browser_compatibility)
+started in Chrome 61, Firefox 60, Edge 16, and Safari 10.1. There are advanced module features, such as import maps, that are not yet
+supported in any browser. Internet Explorer does not have any support for modules.
```

Comment:
FWIW, this is one of the few areas where OpenMRS's use case can make things simpler... it's not an open-to-the-world, customer-facing app, so we can define a minimum spec of browsers to use... I think excluding IE and requiring the browser versions mentioned above is fine.

Reply:
👍 awesome -- I'm actually going to submit an RFC proposal for browser support next. I'll exclude IE.

Reply:
See related #3